### PR TITLE
Ar 509 put date picker to modal as designed

### PR DESCRIPTION
**Pull request by guya**

### CODE REVIEW

File: src/components/Filters/components/Date/FastDayPicker.js [hunk 1]
```diff
@@ -26,14 +27,32 @@ class FastDayPicker extends Component {
 
   static defaultProps = {
     value: null,
+    stringValue: null,
```
Comment (gshilin) [inline src/components/Filters/components/Date/FastDayPicker.js]: move to state

File: src/components/Filters/components/Date/FastDayPicker.js [hunk 2]
```diff
@@ -95,17 +114,49 @@ class FastDayPicker extends Component {
     );
   };
 
+  openPopup = () => this.setState({ isOpen: true });
+
+  closePopup = () => this.setState({ isOpen: false });
+
+  onPopupDayChange = (date) => {
+    const { onDayChange, language } = this.props;
```
Comment (gshilin) [inline src/components/Filters/components/Date/FastDayPicker.js]: move down

File: src/components/Filters/components/Date/FastDayPicker.js [hunk 3]
```diff
@@ -95,17 +114,49 @@ class FastDayPicker extends Component {
...
+    if (date > Date.now())
+      return;
+    this.setState({ stringValue: FastDayPicker.formatDateValue(date, language) });
+    onDayChange(date);
+    this.closePopup();
+  }
+
+  handleDateInputChange = (event, data) => {
+    if (!event) {
```
Comment (gshilin) [inline src/components/Filters/components/Date/FastDayPicker.js]: try to remove it

File: src/components/Filters/components/Date/FastDayPicker.js [hunk 4]
```diff
@@ -95,17 +114,49 @@ class FastDayPicker extends Component {
...
+      return;
+    }
+    const { onDayChange } = this.props;
+    const localeDateFormat = moment.localeData().longDateFormat('L').replace('DD','D').replace('MM','M');
```
Comment (gshilin) [inline src/components/Filters/components/Date/FastDayPicker.js]: move it outside of class

File: src/components/Filters/components/Date/FastDayPicker.js [hunk 5]
```diff
@@ -95,17 +114,49 @@ class FastDayPicker extends Component {
...
+    const day = moment(data.value, localeDateFormat, true);
+    if (day.isValid()) {
+      onDayChange(day.toDate());
+    }
```
Comment (gshilin) [inline src/components/Filters/components/Date/FastDayPicker.js]: put "else" in the same line

File: src/components/Filters/components/Date/FastDayPicker.js [hunk 6]
```diff
@@ -95,17 +114,49 @@ class FastDayPicker extends Component {
...
+    else {
+      this.setState({ stringValue: data.value });
+    }
+  };
+
+  handleKeyDown = () => {
+    this.setState({ isOpen: false });
```
Comment (gshilin) [inline src/components/Filters/components/Date/FastDayPicker.js]: use closePopup()

File: src/stylesheets/includes/_not-sorted-scss.scss [hunk 7]
```diff
@@ -1396,4 +1396,16 @@
 
 .ui.flowing.popup {
   z-index: 10000;
+}
+
+.ui.active.dimmer {
```
Comment (gshilin) [inline src/stylesheets/includes/_not-sorted-scss.scss]: pls remove

File: src/stylesheets/includes/_not-sorted-scss.scss [hunk 8]
```diff
@@ -1396,4 +1396,16 @@
...
+  z-index: 10000;
+}
+
+.ui.dimmer {
```
Comment (gshilin) [inline src/stylesheets/includes/_not-sorted-scss.scss]: - remove
- format file

Comment (guya) [reply src/stylesheets/includes/_not-sorted-scss.scss]: ui.dimmer removed. You do the scss formating

File: src/components/Filters/components/Date/FastDayPicker.js (hunk 4 above)
Comment (guya) [reply src/components/Filters/components/Date/FastDayPicker.js]: What do you mean?

**Pull request merged**